sentry-opentelemetry/lib/sentry/opentelemetry/span_processor.rb

```diff
@@ -0,0 +1,159 @@
+require 'singleton'
+
+module Sentry
+  module OpenTelemetry
+    TraceData = Struct.new(:trace_id, :span_id, :parent_span_id, :parent_sampled, :baggage)
+
+    class SpanProcessor < ::OpenTelemetry::SDK::Trace::SpanProcessor
+      include Singleton
```

Comment:
I made it a singleton because `span_map` needs to be accessed in the propagator and I didn't want to add a global variable.

Reply:
By globally I assume you mean within the threads of the same process? 

When reading the doc, I see

> This means that the hub used in onStart is the same hub used in onEnd and it means that hubs fork properly in async contexts.

But in Ruby SDK, every thread would have their own `Hub`. So I'm not sure if a global `SpanProcessor` can achieve that when it'll probably always use the main hub.

Reply:
I think this part of the spec is outdated, since we no longer use scope to do anything, it shouldn't matter.
@abhiprasad

Reply:
Yup the spec is outdated, will fix in develop!